lib/chat_mailer.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+#
+class DiscourseChat::ChatMailer
+  def self.mail_notifications
+    return unless SiteSetting.chat_enabled
+
+    instance = self.new
+    instance.mail_notifications
+  end
+
+  def mail_notifications
+    while users.exists?
+      users.limit(100).each do |user|
```

Comment:
Is this limit here for performance reasons?

Reply:
Yeah I spoke with David about this, and this was his suggestion on how to make it performant and not a memory issue.